notebook2/notebook/notebook.py

```diff
@@ -497,26 +420,28 @@ def wait_websocket(ws):
     if notebook is None:
         return
 
-    pod_name = notebook['pod_name']
-    svc_name = notebook['svc_name']
+    pod_uuid = notebook['pod_uuid']
+
+    url = external_url_for('')
+    url = f'{url}instance-ready/{pod_uuid}/'
+
     while True:
         try:
-            response = requests.head(f'https://notebook2.hail.is/instance-ready/{svc_name}/',
-                                     timeout=1)
+            response = requests.head(url, timeout=1)
             if response.status_code < 500:
```

Comment:
as a note: we may want to investigate status_code in a more fine grained way. unexpected 404's should be reported

Reply:
Based on comments, something has changed. In particular, status_codes < 500 is no longer an indicator of success. I'm seeing 499's on failure, before the jupyter notebook is ready, both here and in notebook.hail.is. Not sure what has changed.

This method still works, just causes refreshes, which has a side benefit of actually giving finer-grained updates. In any case, I think we should only break on 200.

Reply:
Let's do HEAD and expect a status code of exactly 405, that's how jupyter responds to HEAD on the login page.

Reply:
So I'm seeing 499's (client disconnect) when instance isn't reachable. I'll look into this.